### PR TITLE
feat: add processors crd

### PR DESCRIPTION
**Pull request by blumamir**
This PR introduces the changes in https://github.com/keyval-dev/odigos/pull/978 into the helm chart

### CODE REVIEW

File: charts/odigos/templates/crds/collectorsgroup.yaml [hunk 1]
```diff
@@ -31,8 +31,8 @@ spec:
                 type: string
               role:
                 enum:
-                - GATEWAY
-                - DATA_COLLECTION
+                - CLUSTER_GATEWAY
+                - NODE_COLLECTOR
```
Comment (esara) [inline charts/odigos/templates/crds/collectorsgroup.yaml:35]: I dont think this changed is needed, the crd did not seem to change in the cli installed version
https://github.com/keyval-dev/odigos/blob/main/api/config/crd/bases/odigos.io_collectorsgroups.yaml#L42

File: charts/odigos/templates/crds/processors.yaml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                  type: string
+                type: array
+              type:
+                description: type of the processor (batch, attributes, etc). this
+                  field is only the type, not it's instance name in the collector
+                  configuration yaml
+                type: string
+            required:
+            - collectorRoles
+            - orderHint
```
Comment (esara) [inline charts/odigos/templates/crds/processors.yaml:93]: this is consistent with the cli code
https://github.com/keyval-dev/odigos/blob/main/api/config/crd/bases/odigos.io_processors.yaml#L94
but for some reason, this line is missing from my cli created crd

Comment (blumamir) [reply charts/odigos/templates/crds/processors.yaml:93]: Right! good catch.

There is miss alignment between the auto generated API yaml files [here](https://github.com/keyval-dev/odigos/blob/main/api/config/crd/bases/odigos.io_processors.yaml#L94) and the go code that actually creates the CRDs [here](https://github.com/keyval-dev/odigos/blob/28c7135f631e8d762798f70209fd7a4f00698878/cli/cmd/resources/crds/processor.go#L49).

Will fix this issue in the odigos API module and the helm chart. Thanks for finding this 🙏

File: charts/odigos/templates/crds/collectorsgroup.yaml (hunk 1 above)
Comment (blumamir) [reply charts/odigos/templates/crds/collectorsgroup.yaml:35]: seems I missed updating the value in the [go annotation](https://github.com/keyval-dev/odigos/blob/28c7135f631e8d762798f70209fd7a4f00698878/api/odigos/v1alpha1/collectorsgroup_types.go#L23). 
the values are set correctly in the [cli](https://github.com/keyval-dev/odigos/blob/28c7135f631e8d762798f70209fd7a4f00698878/cli/cmd/resources/crds/collectorsgroup.go#L58)
good catch thanks!

**Comment by blumamir** [comment]
Thanks for the review, very much appreciated 🙏

Fixed the issues you found in https://github.com/keyval-dev/odigos/pull/983

File: charts/odigos/templates/crds/collectorsgroup.yaml (hunk 1 above)
Comment (blumamir) [reply charts/odigos/templates/crds/collectorsgroup.yaml:35]: https://github.com/keyval-dev/odigos/pull/983

File: charts/odigos/templates/crds/processors.yaml (hunk 2 above)
Comment (blumamir) [reply charts/odigos/templates/crds/processors.yaml:93]: https://github.com/keyval-dev/odigos/pull/983

**Pull request merged**